crypto-primitives/src/merkle_tree/tests/mod.rs

```diff
@@ -180,13 +191,154 @@ mod bytes_mt_tests {
+            &serialized_leaves,
+            LeafOrderingMode::BIT_REVERSED,
+        )
+        .unwrap();
+
+        // NOTE: The roots will be DIFFERENT because the physical leaf arrangement is different
+        // This is expected and correct behavior
+        assert_ne!(tree_natural.root(), tree_bit_reversed.root());
+
+        // Test that proofs work correctly for both ordering modes
+        for (i, leaf) in serialized_leaves.iter().enumerate() {
+            // Natural ordering proof
+            let proof_natural = tree_natural.generate_proof(i).unwrap();
+            assert!(proof_natural
+                .verify(
+                    &leaf_crh_params,
+                    &two_to_one_params,
+                    &tree_natural.root(),
+                    leaf.as_slice()
+                )
+                .unwrap());
+
+            // Bit-reversed ordering proof
+            let proof_bit_reversed = tree_bit_reversed.generate_proof(i).unwrap();
+            let actual_leaf = serialized_leaves
+                [bit_reverse_index(i, (tree_bit_reversed.height() - 1) as u32)]
+            .clone();
+            assert!(proof_bit_reversed
+                .verify(
+                    &leaf_crh_params,
+                    &two_to_one_params,
+                    &tree_bit_reversed.root(),
+                    actual_leaf.as_slice()
+                )
+                .unwrap());
+        }
+
+        // Test multi-proofs work with bit-reversed ordering
+        let multi_proof_natural = tree_natural
+            .generate_multi_proof((0..serialized_leaves.len()).collect::<Vec<_>>())
+            .unwrap();
+        assert!(multi_proof_natural
+            .verify(
+                &leaf_crh_params,
+                &two_to_one_params,
+                &tree_natural.root(),
+                serialized_leaves.clone()
+            )
+            .unwrap());
+
+        let multi_proof_bit_reversed = tree_bit_reversed
+            .generate_multi_proof((0..serialized_leaves.len()).collect::<Vec<_>>())
+            .unwrap();
+        assert!(multi_proof_bit_reversed
+            .verify(
+                &leaf_crh_params,
+                &two_to_one_params,
+                &tree_bit_reversed.root(),
+                serialized_leaves.clone()
+            )
+            .unwrap());
+
+        // Verify the ordering mode is stored correctly in multi-proofs
+        assert_eq!(
+            multi_proof_natural.leaf_ordering_mode,
+            LeafOrderingMode::NATURAL
+        );
+        assert_eq!(
+            multi_proof_bit_reversed.leaf_ordering_mode,
+            LeafOrderingMode::BIT_REVERSED
+        );
+    }
+
+    #[test]
+    fn bit_reverse_index_test() {
+        // Test the bit_reverse_index function directly
+        // For height=3 (8 leaves), indices should be:
+        // 0 (000) -> 0 (000)
+        // 1 (001) -> 4 (100)
+        // 2 (010) -> 2 (010)
+        // 3 (011) -> 6 (110)
+        // 4 (100) -> 1 (001)
```

Comment:
So you mean that using ordering BIT_REVERSED the leaves in physical memory locations corresponding to 0 and 4 would get hashed into an internal node (rather than 0 and 1) ?

If so, can you write tests that cover this?